### PR TITLE
Very basic enemy AI

### PR DESCRIPTION
**Pull request by dughiboogie**
There are some classes that are probably unnecessary noise, leftovers of the behaviours modularity. Probably not everything should be merged, but this feels like a solid starting base for enemy AI to merge into development.

### CODE REVIEW

File: scripts/enemy/EnemyFollowEnemyBehaviour.gd [hunk 1]
```diff
@@ -0,0 +1,31 @@
+extends Node3D
+
+
+@onready var enemy: EnemyBase = $".."
```
Comment (MattVsTheWorld) [inline scripts/enemy/EnemyFollowEnemyBehaviour.gd:4]: Is there a better way to reference this? 
Maybe you could export it and assign it directly in the scene
or make the scene unique to reference it

File: scripts/enemy/EnemyFollowPlayerBehaviour.gd [hunk 2]
```diff
@@ -0,0 +1,31 @@
+extends Node3D
+
+
+@onready var enemy: EnemyBase = $".."
```
Comment (MattVsTheWorld) [inline scripts/enemy/EnemyFollowPlayerBehaviour.gd:4]: Same possible reference issue

File: scripts/enemy/EnemySpawner.gd [hunk 3]
```diff
@@ -0,0 +1,16 @@
+extends Node3D
+
+
+@onready var enemies_node = $"../../Enemies"
```
Comment (MattVsTheWorld) [inline scripts/enemy/EnemySpawner.gd:4]: Reference possible issue

**Review by MattVsTheWorld — APPROVED** [review]
Everything seems good, some path references could cause issues but are not immediate issues

File: scripts/enemy/EnemyFollowEnemyBehaviour.gd (hunk 1 above)
Comment (dughiboogie) [reply scripts/enemy/EnemyFollowEnemyBehaviour.gd:4]: This class should always have a EnemyBase script as the root

File: scripts/enemy/EnemySpawner.gd (hunk 3 above)
Comment (dughiboogie) [reply scripts/enemy/EnemySpawner.gd:4]: We need to be careful about the Enemies node. It needs to be present outside the Player node

**Pull request merged**